apps/question-bank/src/main/java/gov/cdc/nbs/questionbank/valueset/ValueSetCreator.java

```diff
@@ -1,82 +1,66 @@
+import gov.cdc.nbs.questionbank.valueset.request.ValueSetCreateRequest;
 import gov.cdc.nbs.questionbank.valueset.response.CreateValueSetResponse;
 import gov.cdc.nbs.questionbank.valueset.util.ValueSetConstants;
-import lombok.RequiredArgsConstructor;
+
 
 @Service
-@RequiredArgsConstructor
 public class ValueSetCreator {
 
-	@Autowired
-	private ValueSetRepository valueSetRepository; // Value Set Repo
-	
-	private CodeValueGeneralRepository codeValueRepository; // ValueSet Concept Repo
-	
-	@Autowired
-	private CodesetGroupMetadatumRepository codeSetGrpMetaRepository; // Value Set Grp Meta
+	private final ValueSetRepository valueSetRepository;
+
+	private final CodesetGroupMetadatumRepository codeSetGrpMetaRepository;
+
+	public ValueSetCreator(ValueSetRepository valueSetRepository,
+			CodesetGroupMetadatumRepository codeSetGrpMetaRepository) {
+		this.valueSetRepository = valueSetRepository;
+		this.codeSetGrpMetaRepository = codeSetGrpMetaRepository;
+	}
 
-	public CreateValueSetResponse createValueSet(ValueSetRequest request, long userId) {
+	public CreateValueSetResponse createValueSet(ValueSetCreateRequest request, long userId) {
 		CreateValueSetResponse response = new CreateValueSetResponse();
-		String codeSetName = request.getValueSetNm().toUpperCase();
-		String codeShrtDescTxt = request.getCodeSetDescTxt();
+		String codeSetName = request.valueSetCode().toUpperCase();
+		String codeShortDescTxt = request.valueSetName();
 
 		if (checkValueSetNameExists(codeSetName)) {
-
 			response.setMessage(ValueSetConstants.VALUE_SET_NAME_EXISTS);
 			response.setStatus(HttpStatus.BAD_REQUEST);
 			return response;
 		}
-		if (checkCodeSetGrpMetaDatEntry(codeShrtDescTxt, codeSetName)) {
+		if (checkCodeSetGrpMetaDatEntry(codeShortDescTxt, codeSetName)) {
 			response.setMessage(ValueSetConstants.CODE_SET_GRP_TEXT_NAME_EXISTS);
 			response.setStatus(HttpStatus.BAD_REQUEST);
 			return response;
 		}
 		try {
-			// Add CodeSet Group MetaData
 			long codeSetGroupID = getCodeSetGroupID();
-			CodeSetGroupMetadatum codeGrp = new  CodeSetGroupMetadatum();
+			CodeSetGroupMetadatum codeGrp = new CodeSetGroupMetadatum();
 			codeGrp.setId(codeSetGroupID);
-			codeGrp.setCodeSetDescTxt(request.getCodeSetDescTxt());
-			codeGrp.setCodeSetNm(request.getValueSetNm());
-			codeGrp.setLdfPicklistIndCd(request.getLdfPicklistIndCd());
-			// Add ValueSet
-			Codeset valueSet = new Codeset(asAdd(request,userId));
-			CodesetId id = new CodesetId();
-			id.setClassCd(ValueSetConstants.CREATE_CLASS_CD);
-			id.setCodeSetNm(codeSetName);
-			valueSet.setId(id);
-			// save changes
+			codeGrp.setCodeSetDescTxt(request.valueSetDescription());
+			codeGrp.setCodeSetShortDescTxt(codeShortDescTxt);
+			codeGrp.setCodeSetNm(codeSetName);
+			codeGrp.setVadsValueSetCode(codeSetName);
+			codeGrp.setLdfPicklistIndCd('Y');
+			codeGrp.setPhinStdValInd('N');   // will add all this properties inside the constructor
```

Comment:
Question: Could the creation of this `CodeSetGroupMetadatum` be moved into the constructor of the `Codeset`? That would put all of the logic for creating the `Codeset` in one place.

Reply:
To create a codeset, a CodeSetGroupMetadatatum must be created first.